File: app/src/main/java/com/raju/javabaseproject/data/source/UserRepository.java
package com.raju.javabaseproject.data.source;

import com.raju.javabaseproject.data.model.User;
import com.raju.javabaseproject.data.source.remote.api.UserApi;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Observable;

public class UserRepository {

    private UserApi api;

    @Inject
    UserRepository(UserApi api) {
        this.api = api;
    }

    public Observable<List<User>> getUsers() {
        return api.getUsers();
    }
}

File: app/src/main/java/com/raju/javabaseproject/data/source/UserRepository.kt
package com.raju.javabaseproject.data.source

import com.raju.javabaseproject.data.model.User
import com.raju.javabaseproject.data.source.remote.api.UserApi

import javax.inject.Inject

import io.reactivex.Observable

class UserRepository @Inject
    internal constructor(private val api: UserApi) {

    val users: Observable<MutableList<User>>
        get() = api.users
}
